src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -6122,6 +6114,750 @@ .maxstack  3
+                        {
+                            Console.Write("{0}, ", i);
+                        }
+                    }
+                    static IEnumerable<int> F(int x, int y, IEnumerable<int> e) => [x, y, ..e];
+                }
+                """;
+            CompileAndVerify(
+                source,
+                targetFramework: TargetFramework.Net80,
+                verify: Verification.FailsPEVerify,
+                expectedOutput: IncludeExpectedOutput("1, 2, 3, "));
+        }
+
+        // Compare members of synthesized types to a similar type from source.
+        [Fact]
+        public void SynthesizedReadOnlyList_Members()
+        {
+            string sourceA = """
+                using System;
+                using System.Collections;
+                using System.Collections.Generic;
+                internal sealed class ReadOnlyArray<T> :
+                    IEnumerable<T>,
+                    IReadOnlyCollection<T>,
+                    IReadOnlyList<T>,
+                    ICollection<T>,
+                    IList<T>
+                {
+                    private readonly T[] _items;
+                    public ReadOnlyArray(T[] items) { _items = items; }
+                    IEnumerator IEnumerable.GetEnumerator() => _items.GetEnumerator();
+                    IEnumerator<T> IEnumerable<T>.GetEnumerator() => throw new NotSupportedException();
+                    int IReadOnlyCollection<T>.Count => _items.Length;
+                    T IReadOnlyList<T>.this[int index] => _items[index];
+                    int ICollection<T>.Count => _items.Length;
+                    bool ICollection<T>.IsReadOnly => true;
+                    void ICollection<T>.Add(T item) => throw new NotSupportedException();
+                    void ICollection<T>.Clear() => throw new NotSupportedException();
+                    bool ICollection<T>.Contains(T item) => throw new NotSupportedException();
+                    void ICollection<T>.CopyTo(T[] array, int arrayIndex) { }
+                    T IList<T>.this[int index]
+                    {
+                        get => _items[index];
+                        set => throw new NotSupportedException();
+                    }
+                    int IList<T>.IndexOf(T t) => Array.IndexOf<T>(_items, t);
+                    void IList<T>.Insert(int index, T item) => throw new NotSupportedException();
+                    bool ICollection<T>.Remove(T item) => throw new NotSupportedException();
+                    void IList<T>.RemoveAt(int index) => throw new NotSupportedException();
+                }
+                """;
+            string sourceB = """
+                using System.Collections.Generic;
+                class Program
+                {
+                    static void Main()
+                    {
+                        IEnumerable<int> x = [0];
+                        IEnumerable<int> y = [..x];
```

Comment:
Might be helpful to add a comment indicating that these are just here to ensure the synthesized readonly types get synthesized.